internal/db/schema/migrations/oss/postgres/101/01_app_token_base.up.sql

```diff
@@ -0,0 +1,246 @@
+          old.revoked, new.revoked;
+      end if;
+    end if;
+
+    return new;
+  end;
+  $$ language plpgsql;
+  comment on function validate_app_token_revocation() is
+    'validate_app_token_revocation ensures that the revoked field can only be updated from false to true, '
+    'preventing tokens from being un-revoked or other invalid state transitions';
+
+  -- App token deleted tracking table
+  create table app_token_deleted (
+    public_id wt_public_id primary key,
+    delete_time wt_timestamp not null
+  );
+  comment on table app_token_deleted is
+    'app_token_deleted holds the ID and delete_time of every deleted app token. '
+    'It is automatically trimmed of records older than 30 days by a job.';
+
+  -- Create trigger for app_token deletion tracking
+  create trigger insert_deleted_id after delete on app_token
+    for each row execute function insert_deleted_id('app_token_deleted');
+
+  -- Create index on delete_time for app_token_deleted
+  create index app_token_deleted_delete_time_idx on app_token_deleted (delete_time);
+
+  create table app_token_permission (
+    private_id wt_private_id primary key,
+    app_token_id wt_public_id not null,
+      constraint app_token_permission_app_token_id_idx
+        foreign key (app_token_id) references app_token(public_id)
+        on delete cascade
+        on update cascade
+  );
+  comment on table app_token_permission is
+    'app_token_permission is the base table for application token permissions.';
+
+  -- Create index on app_token_id for efficient token-based lookups
+  create index app_token_permission_app_token_id_idx on app_token_permission (app_token_id);
+
+  create function insert_app_token_permission_subtype() returns trigger
+  as $$
+  begin
+    insert into app_token_permission
+      (private_id, app_token_id)
+    values
+      (new.private_id, new.app_token_id);
+    return new;
+  end;
+  $$ language plpgsql;
+  comment on function insert_app_token_permission_subtype() is
+    'insert_app_token_permission_subtype is used to automatically insert a row into the app_token_permission table '
+    'whenever a row is inserted into the subtype table';
+
+  create table app_token_permission_grant (
+    permission_id wt_private_id
+      constraint app_token_permission_grant_fkey
+      references app_token_permission(private_id)
+        on delete cascade
+        on update cascade,
+    canonical_grant wt_canonical_grant not null
+      constraint app_token_permission_iam_grant_fkey
+        references iam_grant(canonical_grant)
+        on delete cascade
+        on update cascade,
+    raw_grant text not null
+      constraint raw_grant_must_not_be_empty
+      check(
+        length(trim(raw_grant)) > 0
+      ),
+    primary key(permission_id, canonical_grant)
+  );
+  comment on table app_token_permission_grant is
+    'app_token_permission_grant contains grants assigned to app tokens in project scope';
```

Comment:
It looks like this table can apply to app tokens at any scope, not just the project scope. Should this instead say,

`'app_token_permission_grant contains grants assigned to app tokens'`